### ISSUE TITLE
HTTP3 DHT

### ISSUE THREAD

**Issue by milahu** [webtorrent/webtorrent#2877]
HTTP/3 is based on UDP.
will this allow us to connect to DHT and UDP trackers?
will this work over IPv6?

possible alternative to #288 and #1492

related: https://github.com/webtorrent/webtorrent/issues/288#issuecomment-147548619

> QUIC/UDP DataChannels (https://www.youtube.com/watch?v=mIvyOFu1c1Q) will make a WebRTC DHT a lot more efficient, as it has a 0RTT protocol for p2p connections, vs the 4RTT one used by SCTP/DTLS/UDP Data Channels today

browser support: [caniuse.com/http3](https://caniuse.com/http3)

what would IPFS do: [docs.libp2p.io/concepts/transports/quic/](https://docs.libp2p.io/concepts/transports/quic/)

> QUIC is a new transport protocol that provides an always-encrypted, stream-multiplexed connection built on top of UDP.

> In addition to defining the QUIC transport, the IETF also standardized a new version of HTTP that runs on top of QUIC

i guess that "always-encrypted" breaks our use case (?)

**Comment by SilentBot1** [comment]
> will this allow us to connect to DHT and UDP trackers?
No. You also can't connect to the uTP/TCP addresses returned in the browser even if you could.

> will this work over IPv6?
Yes, just like HTTP2, WebRTC or WebSockets.

WebTransport/QUIC could be used as another data channel like WebRTC from browsers, but the buy-in from other clients would reset to 0.

WebTransport/QUIC can also not be used to host servers from the browser, so you could never create a browser-only DHT, without a supporting server or non-browser-based peer.

**Comment by milahu** [comment]
ok, so we need [web browsers with UDP support](https://github.com/Fyrd/caniuse/issues/7198)
